interview-questions.md

```diff
@@ -10,40 +10,102 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: Floats are floating point numbers. This means that they allow the decimal to move within their value, providing a wide range of useful values with a limited number of significant figures. Integers are whole numbers. Ruby implementation of these classes is signed, meaning that a bit is used to indicate whether the value is positive or negative. Ruby further provides methods for these classes both directly and through duck typing.
 
-Researched answer:
+Researched answer: It took some research to find the exact ranges for Ruby's implementation of these classes. Specifically for numbers, which represent integers, two subclasses are relevant. fixnum represents normal integers used by machine logic. bignum represents integers beyond this normal range. I suspect that under the hood, the code is referencing multiple memory locations to accomplish this. 
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: Implicit return means that the final line of code in a statement is used to determine the return from the block. For contrast, explicit return requires a return statement to identify what should exit from that block. In Ruby, the use of implicit return prevents compiler errors from unexpected undefined values being returned from blocks (as JavaScript performs). Float is a separate subclass that inherits/implements the numeric class. The representation of floats within Ruby presents several changes with rounding errors between decimal and binary values. Backwards error analysis can provide better understanding for the number of significant figures maintained in computations involving these floating point numbers. Rounding errors due to floating point value inaccuracy led to the initial basis for the butterfly effect theory.
 
 Researched answer:
 
+Ruby implicit returns can provide some undesired behavior when programmers do not consider the return provided by function or method calls. It can also provide tricky interaction when using console commands within loops and functions. For example, iterating backwards through a loop would provide very different return with a p or puts command within the loop compared to running the p or puts after the function.
+
+```ruby
+my_array = [1, 2, 3]
+for i in my_array
+    p i
+end
+
+# 1
+# 2
+# 3
+
+output = for i in my_array
+    i
+end
+
+p output
+
+# [1, 2, 3]
+
+```
+
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: Ruby blocks group together operations within a unanamous function.
```

Comment:
This definitely describes a block, there is a lot of room to expand on this answer though.

Reply:
I think by this point in the questions, I didn't really have a firm enough understanding of how a Ruby block was distinct from a code block. I could have discussed both, but opted to dive deeper for my researched answer.